govcd/adminorg_test.go

```diff
@@ -0,0 +1,38 @@
+// +build org functional ALL
+
+/*
+ * Copyright 2019 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+package govcd
+
+import (
+	"fmt"
+	. "gopkg.in/check.v1"
+)
+
+// Creates a Catalog and then verify that finds it
+func (vcd *TestVCD) Test_Client_FinAdminCatalogByName(check *C) {
+	if vcd.skipAdminTests {
+		check.Skip(fmt.Sprintf(TestRequiresSysAdminPrivileges, check.TestName()))
+	}
+
+	adminOrg, err := vcd.client.GetAdminOrgByName(vcd.config.VCD.Org)
+	check.Assert(err, IsNil)
+	check.Assert(adminOrg, NotNil)
+	catalogName := "catalogForQuery"
+	adminCatalog, err := adminOrg.CreateCatalog(catalogName, "catalogForQueryDescription")
+	check.Assert(err, IsNil)
+	AddToCleanupList(catalogName, "catalog", vcd.config.VCD.Org, check.TestName())
+	check.Assert(adminCatalog.AdminCatalog.Name, Equals, catalogName)
+
+	// just imitate wait
+	err = adminOrg.Refresh()
+	check.Assert(err, IsNil)
+
+	findRecords, err := adminOrg.FinAdminCatalogByName(catalogName, adminOrg.AdminOrg.Name)
+	check.Assert(err, IsNil)
+	check.Assert(findRecords, NotNil)
+	check.Assert(len(findRecords), Equals, 1)
+	check.Assert(findRecords[0].Name, Equals, catalogName)
+	check.Assert(findRecords[0].OrgName, Equals, adminOrg.AdminOrg.Name)
```

Comment:
This test proves that the function finds a catalog, but not that the catalog is shared, which is the point of this PR.
Is there a way of testing the shared catalog scenario ?

Reply:
I don't care about it shared, I care about if catalog belongs same ORG. Fully verify unit test would create two orgs, catalogs and make one of shared for other Orgs. And when delete  Org and expects it didn't touch catalog not belonged to other org.

Reply:
What I mean is: do we have a way of testing a negative finding? Perhaps using one of the tests where we create other catalogs.

Reply:
we can't create shared catalog, we can only find not existing one